src/AssetsDetectionController.ts

```diff
@@ -63,12 +63,17 @@ export class AssetsDetectionController extends BaseController<AssetsDetectionCon
 		const { selectedAddress } = this.config;
 		const api = this.getOwnerCollectiblesApi(selectedAddress);
 		const assetsController = this.context.AssetsController as AssetsController;
-		let response;
-		/* istanbul ignore if */
-		if (assetsController.openSeaApiKey) {
-			response = await fetch(api, { headers: { 'X-API-KEY': assetsController.openSeaApiKey } });
-		} else {
-			response = await fetch(api);
+		let response: Response;
+		try {
+			/* istanbul ignore if */
+			if (assetsController.openSeaApiKey) {
+				response = await timeoutFetch(api, { headers: { 'X-API-KEY': assetsController.openSeaApiKey } }, 2000);
```

Comment:
2 seconds it's too low.  I'd go with at least 10-15. Remember the app might be used with not so great internet connections, which is usually not a problem in desktop.

Also move it to a constant at the top of the file specifying what it is for.

Reply:
good point, totally agree